### DISCUSSION TITLE
Livewire and Laravel Echo using pusher

### DISCUSSION THREAD

**Discussion by sagadsalem** [livewire/livewire#7787] [7. Bugs (archived)]
### Livewire version

3.1

### Laravel version

10.10

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

I am trying to do some real-time stuff
I have an e-Commerce app that when I receive an order I notify the merchant that he has a new order
1- I Create an Event
php artisan make:event NewOrderNotification
```
<?php

namespace App\Events;

use Illuminate\Queue\SerializesModels;
use Illuminate\Broadcasting\PrivateChannel;
use Illuminate\Foundation\Events\Dispatchable;
use Illuminate\Broadcasting\InteractsWithSockets;
use Illuminate\Contracts\Broadcasting\ShouldBroadcast;

class NewOrderNotification implements ShouldBroadcast
{
    use Dispatchable;
    use InteractsWithSockets;
    use SerializesModels;

    public function __construct(public int $orderId)
    {
        //
    }

    public function broadcastOn(): array
    {
        return [
            new PrivateChannel('App.Models.Order.' . $this->orderId),
        ];
    }
}

```

and set my authentication
```
Broadcast::channel('App.Models.Order.{order}', function (Client $client, Order $order) {
    return $client->id === $order->client_id;
});
```

and in my Livewire Component, i have
```
  #[On('echo-private:App.Models.Order.{order},NewOrderNotification')]
    public function orderNofication()
    {

    }
```

and i get this error

Unable to evaluate dynamic event name placeholder: {order}

### Repository or Wirebox URL

https://github.com/

### Additional information

I think i provide everything in the description and there are no need the repo link

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by gazzoy** [comment]
Did you try `getListeners()` instead of `On` attribute as below?

```
    public function getListeners()
    {
        return [
            "echo-private:App.Models.Order.{$this->order},NewOrderNotification', => 'orderNofication',
        ];
    }

    public function orderNofication()
    {
    }

```

**Reply by sagadsalem** [reply]
No, but I solve the problem because I forget to define my public Order in the Component that received  the event

**Reply by gazzoy** [reply]
Glad to here that.
